3rdparty/openvx/include/openvx_hal.hpp

```diff
@@ -1018,6 +1036,32 @@ inline int ovx_hal_cvtOnePlaneYUVtoBGR(const uchar * a, size_t astep, uchar * b,
     return CV_HAL_ERROR_OK;
 }
 
+inline int ovx_hal_integral(int depth, int sdepth, int , const uchar * a, size_t astep, uchar * b, size_t bstep, uchar * c, size_t , uchar * d, size_t , int w, int h, int cn)
+{
+    if (depth != CV_8U || sdepth != CV_32S || c != NULL || d != NULL || cn != 1)
+        return CV_HAL_ERROR_NOT_IMPLEMENTED;
+
+    try
+    {
+        vxContext * ctx = vxContext::getContext();
+        vxImage ia(*ctx, a, astep, w, h);
+        vxImage ib(*ctx, (uint *)(b+bstep+sizeof(uint)), bstep, w, h);
```

Comment:
Since OpenVX support both U32 and S32 types IMO wrappers should handle `int` and `unsigned` data as different OpenVX image types for a generic case. But for integral image OpenVX support only 8U->32U processing, while opencv doesn't use 32U type, so I've casted type here to retain more generic wrapper.